Parser/Device/AbstractDeviceParser.php

```diff
@@ -2297,13 +2299,11 @@ protected function parseClientHints(): ?array
             $deviceType  = null;
             $formFactors = $this->clientHints->getFormFactors();
 
-            if (\count($formFactors) > 0) {
-                foreach (self::$clientHintFormFactorsMapping as $formFactor => $deviceTypeId) {
-                    if (\array_key_exists($formFactor, $formFactors)) {
-                        $deviceType = self::getDeviceName($deviceTypeId);
+            foreach ($formFactors as $formFactor) {
+                if (isset(self::$clientHintFormFactorsMapping[$formFactor])) {
+                    $deviceType = self::$clientHintFormFactorsMapping[$formFactor];
 
-                        break;
-                    }
+                    break;
                 }
             }
```

Comment:
```php
            foreach (self::$clientHintFormFactorsMapping as $formFactor => $deviceTypeId) {
                if (\in_array($formFactor, $formFactors)) {
                    $deviceType = $deviceTypeId;

                    break;
                }
            }
```

Reply:
I get that we need to consider priority. But if the header contains `"EInk", "Watch"` or `"Desktop", "EInk"`, we'll recognize it as a tablet. Should we maybe drop the priority for eink?

Reply:
You can omit it, after desktop.

Reply:
Thank you. I have updated the priorities and modified the fixture so that it will catch changing that algorithm to my previous version or the sorting of $clientHintFormFactorsMapping